### DISCUSSION TITLE
api-auth login with curl gives CSRF verification failed

### DISCUSSION THREAD

**Discussion by Dan1ell** [mediacms-io/mediacms#253] [Q&A]
How to log into mediacms with the rest api?
I tested on the demo site as follows:

curl -X POST -d '{"username": "myusername", "password": "mypassword"}' -H 'Content-Type: application/json' https://demo.mediacms.io/api-auth/login/ 

But instead of the token I get CSRF verification failed.

**Comment by mgogoulos** [comment]
Hi, 

both urls
https://demo.mediacms.io/api-auth/login/ and https://demo.mediacms.io/accounts/login/

are Django forms and not REST API endpoints, so you cannot POST with username/password only but instead have to render the full form.

I'm ready to open a PR that will include a few bacic urls that are missing (example a /whoami API showing info about the request, /api/login to authenticate and return a TOKEN, plus a few more). I'll make sure I add them on the Swagger documentation too  - https://demo.mediacms.io/swagger/


Until then, maybe you may achieve what you're trying by providing the user/password credentials on each call, for example:

```
curl --user usernamex:passwordy https://demo.mediacms.io/api/v1/manage_comments -H "Accept: application/json" 
```

**Reply by Dan1ell** [reply]
Awesome, about a PR being underway. I'm starting a native mobile app so a login token would be required.

**Comment by mgogoulos** [comment]
@Dan1ell I've just merged https://github.com/mediacms-io/mediacms/pull/257 which adds these URLS, checkout on description and let us know if they work for you!
